### PR TITLE
fix(patreon): fix router regression

### PR DESCRIPTION
**Pull request by evakill**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description
fixes a regression introduced in the [react router v6 upgrade](https://github.com/ONEARMY/community-platform/pull/3069/files#diff-9bd14857524cac55f2458432800e60e67b5056dddeca9857127f413a3c9232b9) which was causing a whitescreen on patreon redirect

_What this PR does_

## Git Issues

Closes #

## Screenshots/Videos

_If useful, provide screenshot or capture to highlight main changes_

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of our monthly maintainers call (first Monday of the month)

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

File: src/pages/UserSettings/SettingsPage.tsx [hunk 1]
```diff
@@ -312,7 +312,7 @@ export const SettingsPage = observer((props: IProps) => {
                     />
                   )}
                   <AuthWrapper roleRequired={'beta-tester'}>
-                    <PatreonIntegration user={user} />
```
Comment (evakill) [inline src/pages/UserSettings/SettingsPage.tsx:315]: this was added [here](https://github.com/ONEARMY/community-platform/pull/3099/files#diff-dea43f99ef928d63f929338d6ee3e5c0f9abb36c629afb8fe702b1f7048fe21eR315) but the  `PatreonIntegration` component was not updated to accept this prop

**Review by iSCJT — APPROVED** [review]

**Review by benfurber — APPROVED** [review]

File: src/pages/UserSettings/SettingsPage.tsx (hunk 1 above)
Comment (thisislawatts) [reply src/pages/UserSettings/SettingsPage.tsx:315]: **issue(blocking):** @evakill can we introduce a tests that would have caught this failure?

**Review by thisislawatts — DISMISSED** [review]

File: src/pages/UserSettings/SettingsPage.tsx (hunk 1 above)
Comment (evakill) [reply src/pages/UserSettings/SettingsPage.tsx:315]: @thisislawatts i am not sure how this made it through; my local env caught it as a TS error. maybe something is broken in our TS config? don't think this should block this PR though

Comment (iSCJT) [reply src/pages/UserSettings/SettingsPage.tsx:315]: I've been finding stuff the past couple of days which I'm not sure how it got through either as my local env was throwing TS errors.  Also related to the router upgrade.

Comment (mariojsnunes) [reply src/pages/UserSettings/SettingsPage.tsx:315]: Before the router upgrade there were already some TS errors (ex: stores) so these might have been overlooked, sorry!

Comment (evakill) [reply src/pages/UserSettings/SettingsPage.tsx:315]: all good @mariojsnunes!! as @thisislawatts points out, regressions reflect more on our testing than the introduced code ;) 

plus i am not sure where the TS stuff came from; it know that i've seen it before in other files as well

**Pull request merged**